iAM-legacy/Simulation/AppliedTreatment.cs

```diff
@@ -24,6 +24,10 @@ public class AppliedTreatment
         String m_strChangeHash;
         AppliedTreatment _multipleTreatment;//Stores a follow up treatment.
         bool _isExclusive;
+        //For split treatment (before split).
```

Comment:
The `AppliedTreatments` class is violating encapsulation by making its properties publicly settable. The class needs to be changed to have the properties passed as parameters to the constructor and all of the public `set`'s should be made private.